test/unit/persist.spec.js

```diff
@@ -1117,5 +1117,46 @@ describe("Persist", () => {
 			expect(length4).to.be.equals(3);
 			expect(length5).to.be.equals(3);
 		});
+		it("should check remove 'b' depth when replaceDepth for start -> a -> b -> c(replace)", () => {
+			// Given
+			const mockModules = injectPersistModules();
+			const mockHistory = mockModules["./browser"].history;
+			const InjectedPersist = injectPersistExports(mockModules);
+			const persist = new InjectedPersist.default("");
+
+			InjectedPersist.default.clear();
+
+			// When
+			// start
+			persist.set("0", 1);
+
+			// a
+			mockHistory.pushState({}, "", "/a");
+
+			persist.set("a", 1);
+
+			// b
+			mockHistory.pushState({}, "", "/b");
+
+			persist.set("b", 1);
+
+			// c (replace)
+			mockHistory.replaceState({}, "", "/c");
+
+			InjectedPersist.replaceDepth();
+
+
+			// start -> a(x) -> c
+			const depths = StorageManager.getStateByKey(CONST_PERSIST_STATE, CONST_DEPTHS);
+
+			// Then
+			expect(depths.length).to.be.equals(3);
+			// start
+			expect(depths[0].lastIndexOf(pathname)).to.be.equals(depths[0].length - pathname.length);
+			// a
+			expect(depths[1].lastIndexOf("a")).to.be.equals(depths[1].length - 1);
+			// c
+			expect(depths[2].lastIndexOf("c")).to.be.equals(depths[2].length - 1);
```

Comment:
Testing like this does not look secure I think.
Like, it looks like `depths[1]` is some URL like `~~~/a` I guess,
but when `depths[1]` is just an empty string(`""`) the test will pass without any warning.

What about using a simple regex for testing them? Like, `/a$/.test(depths[1])` or something.